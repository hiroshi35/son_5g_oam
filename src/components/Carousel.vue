<template>
  <div :id="id" class="carousel slide" data-ride="carousel">

    <ol class="carousel-indicators">
      <li v-for="(item, index) in items" :data-target="'#'+id" :data-slide-to="index"
          :class="{ 'active': index === 0 }"></li>


    </ol>
    <div class="carousel-inner" role="listbox">
      <div v-for="(item, index) in items" class="item" :class="{ 'active': index === 0 }">
        <img :src="item.imgSrc">

        <div class="carousel-caption">
          {{item.caption}}
        </div>
      </div>


    </div>
    <a class="left carousel-control" :href="'#'+id" role="button" data-slide="prev">
      <span class="glyphicon glyphicon-chevron-left" aria-hidden="true"></span>
      <span class="sr-only">Previous</span>
    </a>
    <a class="right carousel-control" :href="'#'+id" role="button" data-slide="next">
      <span class="glyphicon glyphicon-chevron-right" aria-hidden="true"></span>
      <span class="sr-only">Next</span>
    </a>
    <slot></slot>
  </div>
</template>

<script>
  export default {
    props: {
      id: {
        required: true
      }
    },
    data () {
      return {
        items: []
      }
    }
  }
</script>
