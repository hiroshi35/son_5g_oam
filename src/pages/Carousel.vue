<template>
  <div>
    <div class="container-fluid">
      <page-header title="Carousel"></page-header>
      <div class="row">
        <div class="col-md-12">
          <panel type="panel-white right-angular">
            <h4 slot="header">Bootstrap Carousel |
              <small>A slideshow component for cycling through elements</small>
            </h4>
            <carousel id="myCarousel">
              <carousel-item img-src="/static/imgs/carousel/1.jpg" caption="First Image"></carousel-item>
              <carousel-item img-src="/static/imgs/carousel/2.jpg" caption="Second Image"></carousel-item>
              <carousel-item img-src="/static/imgs/carousel/3.jpg" caption="Third Image"></carousel-item>
            </carousel>
          </panel>
        </div>
      </div>
    </div>
    <!-- /.container-fluid -->

  </div>
</template>

<script>
  import Carousel from '../components/Carousel.vue'
  import CarouselItem from '../components/CarouselItem.vue'

  export default {
    components: {
      Carousel,
      CarouselItem
    }
  }
</script>