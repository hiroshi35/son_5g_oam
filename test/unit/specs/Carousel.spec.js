import { shallow } from 'vue-test-utils'
import Carousel from '@/components/Carousel.vue'

describe('Carousel.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(Carousel, {
      propsData: {
        id: 'carousel'
      }
    })
    wrapper.setData({items: [{imgSrc: '/static/imgs/carousel/1.jpg', caption: 'Image'}]})
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('renders correct tab id', () => {
    expect(wrapper.contains('.carousel-inner')).toBeTruthy()
  })
})
