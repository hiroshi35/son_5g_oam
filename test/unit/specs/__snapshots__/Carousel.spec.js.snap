// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel.vue has the expected html structure 1`] = `
<div
  class="carousel slide"
  data-ride="carousel"
  id="carousel"
>
  <ol
    class="carousel-indicators"
  >
    <li
      class="active"
      data-slide-to="0"
      data-target="#carousel"
    />
  </ol>
   
  <div
    class="carousel-inner"
    role="listbox"
  >
    <div
      class="item active"
    >
      <img
        src="/static/imgs/carousel/1.jpg"
      />
       
      <div
        class="carousel-caption"
      >
        
        Image
      
      </div>
    </div>
  </div>
   
  <a
    class="left carousel-control"
    data-slide="prev"
    href="#carousel"
    role="button"
  >
    <span
      aria-hidden="true"
      class="glyphicon glyphicon-chevron-left"
    />
     
    <span
      class="sr-only"
    >
      Previous
    </span>
  </a>
   
  <a
    class="right carousel-control"
    data-slide="next"
    href="#carousel"
    role="button"
  >
    <span
      aria-hidden="true"
      class="glyphicon glyphicon-chevron-right"
    />
     
    <span
      class="sr-only"
    >
      Next
    </span>
  </a>
   
</div>
`;
